packages/eslint-plugin-ts/rules/quotes/quotes.test.ts

```diff
@@ -303,11 +303,11 @@ ruleTester.run('quotes', rule, {
 
     // `backtick` should not warn import/export sources.
     {
-      code: 'import "a"; import \'b\';',
+      code: 'import "a" assert { type: "json" }; import \'b\' assert { type: \'json\' };',
       options: ['backtick'],
     },
     {
-      code: 'import a from "a"; import b from \'b\';',
+      code: 'import a from "a" assert { type: "json" }; import b from \'b\' assert { type: \'json\' };',
```

Comment:
Could you add a new test case for `assert` instead of modifying the original two? Thanks